lib/src/boilerplate_suggestors/advanced_props_and_state_class_migrator.dart

```diff
@@ -28,11 +29,51 @@ import 'boilerplate_utilities.dart';
 class AdvancedPropsAndStateClassMigrator extends GeneralizingAstVisitor
     with AstVisitingSuggestorMixin
     implements Suggestor {
+  final bool shouldMigrateCustomClassAndMixins;
+
+  AdvancedPropsAndStateClassMigrator(
+      {this.shouldMigrateCustomClassAndMixins = false});
+
   @override
-  visitClassDeclaration(ClassDeclaration node) {
+  void visitClassDeclaration(ClassDeclaration node) {
     super.visitClassDeclaration(node);
 
     if (!shouldMigrateAdvancedPropsAndStateClass(node)) return;
+
+    final extendsFromCustomClass = !extendsFromUiPropsOrUiState(node);
+    final hasMixins = node.withClause != null;
+
+    // Don't operate if the props class uses mixins and extends a custom class,
+    // unless the flag has been set.
+    if (hasMixins &&
+        extendsFromCustomClass &&
+        !shouldMigrateCustomClassAndMixins) return;
+
+    final className = stripPrivateGeneratedPrefix(node.name.toSource());
+    var newClassDeclarationString =
+        '\n\nclass $className = Ui${className.contains('Props') ? 'Props' : 'State'} with ';
+
+    if (extendsFromCustomClass) {
+      final parentClass = node.extendsClause.superclass.toSource() + 'Mixin';
+      newClassDeclarationString +=
+          '$parentClass, ${className}Mixin${hasMixins ? ',' : ';'}';
+    }
+
+    if (hasMixins) {
+      if (!extendsFromCustomClass) {
+        newClassDeclarationString += '${className}Mixin,';
+      }
+
+      newClassDeclarationString = node.withClause?.childEntities
```

Comment:
This shouldn't have null-aware operators since hasMixins is true, meaning it will never be null. Also, if it were null, `newClassDeclarationString` would get assigned to `null`.